src/System.Runtime.Extensions/src/System/Runtime/Versioning/FrameworkName.cs

```diff
@@ -62,19 +62,24 @@ public String FullName
             {
                 if (_fullName == null)
                 {
-                    StringBuilder sb = new StringBuilder();
-                    sb.Append(Identifier);
-                    sb.Append(c_componentSeparator);
-                    sb.Append(c_versionKey).Append(c_keyValueSeparator);
-                    sb.Append(c_versionValuePrefix);
-                    sb.Append(Version);
-                    if (!String.IsNullOrEmpty(Profile))
+                    if (string.IsNullOrEmpty(Profile))
                     {
-                        sb.Append(c_componentSeparator);
-                        sb.Append(c_profileKey).Append(c_keyValueSeparator);
-                        sb.Append(Profile);
+                        // Roslyn will optimize this to call string.Concat, avoiding any intermediary allocations
+                        _fullName =
+                            Identifier +
+                            c_componentSeparator + c_versionKey + c_keyValueSeparator + c_versionValuePrefix +
+                            Version.ToString();
+                    }
+                    else
+                    {
+                        // A string array is allocated here for params, but intermediary strings are still avoided
```

Comment:
Nit: I don't think these comments are necessary (there are tons of places we use string concatenation without describing what the compiler does).